src/bindings/scala/api/src/main/scala/com/ctc/omega_edit/SessionImpl.scala

```diff
@@ -89,13 +89,17 @@ private[omega_edit] class SessionImpl(p: Pointer, i: FFI) extends Session {
     Edit(i.omega_edit_overwrite(p, offset, s, 0))
 
   /** omega_edit_undo_last_change returns the *negative* serial number of the
-    * change, so negate it to use same success/fail logic.
+    * change, so perform different matching for change id
     *
     * @see
     *   https://github.com/ctc-oss/omega-edit/wiki#undo
     */
   def undoLast(): Result =
-    Edit(i.omega_edit_undo_last_change(p))
+    i.omega_edit_undo_last_change(p) match {
+      case 0          => Change.Paused
+      case v if v > 0 => Change.Fail
+      case v          => Changed(v)
```

Comment:
I'm not really sure what's going on here.  A zero result from undo is an error (it could be due to session edits being paused, or it could be due to trying to undo from an empty active edit stack, etc.), and a negative result is a success.  There are no positive results from undo.

Reply:
So from Adam says it is because of `omega_edit_undo_last_change returns the *negative* serial number of the change`. So it for this method a different check needs done because of how the value is being returned

Reply:
If we don't negate or do this different matching then undo fails in places it should work, your first comment proves this as you tested the change made to remove negate and do no different matching.

Reply:
I get that matching needs to be different because of the sign change.  The matching looks odd to me because I'd think we'd want:

```Scala
case 0  => Change.Fail
case v  => Changed(v)
```

The return is either 0 (a failure) or a negative change ID.

Reply:
Okay so that will still be different than the default I will make it be this